src/Build/BackEnd/Components/SdkResolution/MainNodeSdkResolverService.cs

```diff
@@ -78,148 +57,57 @@ public override void ClearCaches()
             {
-                case NodePacketType.ResolveSdkRequest:
-                    HandleRequest(node, packet as SdkResolverRequest);
-                    break;
-            }
-        }
-
-        /// <inheritdoc cref="ISdkResolverService.ResolveSdk"/>
-        public override SdkResult ResolveSdk(int submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, string solutionPath, string projectPath, bool interactive, bool isRunningInVisualStudio)
-        {
-            ErrorUtilities.VerifyThrowInternalNull(sdk, nameof(sdk));
-            ErrorUtilities.VerifyThrowInternalNull(loggingContext, nameof(loggingContext));
-            ErrorUtilities.VerifyThrowInternalNull(sdkReferenceLocation, nameof(sdkReferenceLocation));
-            ErrorUtilities.VerifyThrowInternalLength(projectPath, nameof(projectPath));
-
-            return _cachedSdkResolver.ResolveSdk(submissionId, sdk, loggingContext, sdkReferenceLocation, solutionPath, projectPath, interactive, isRunningInVisualStudio);
-        }
-
-        /// <summary>
-        /// Handles a request from a remote node.
-        /// </summary>
-        /// <param name="node">The ID of the remote node.</param>
-        /// <param name="request">The <see cref="SdkResolverRequest"/> containing information about the SDK to resolve.</param>
-        /// <remarks>This method must not directly handle requests because it would block requests from other nodes.  Instead, it simply
-        /// adds requests to a queue which are processed by a background thread.</remarks>
-        private void HandleRequest(int node, SdkResolverRequest request)
-        {
-            if (_requestHandler == null)
-            {
-                // Start the background thread which will process queued requests if it has not already been started.
-                lock (_lockObject)
-                {
-                    if (_requestHandler == null)
-                    {
-                        // Create the event used to signal that a request was received
-                        _requestReceivedEvent = new ManualResetEvent(initialState: false);
-
-                        // Create the queue used to store requests that need to be processed
-                        _requests = new ConcurrentQueue<SdkResolverRequest>();
-
-                        // Create the thread which processes requests
-                        _requestHandler = Task.Factory.StartNew(RequestHandlerPumpProc, TaskCreationOptions.LongRunning);
-                    }
-                }
+                return;
             }
 
             // Associate the node with the request
             request.NodeId = node;
 
-            _requests.Enqueue(request);
+            Task.Run(() =>
+            {
+                Thread.CurrentThread.Name = $"Process SDK request {request.Name} for node {request.NodeId}";
```

Comment:
Renaming a ThreadPool thread feels fishy to me. This task will only be running briefly.

Reply:
After a thread is returned to the ThreadPool, I would assume no one cares about it, right? The previous version renamed a newly-created thread, which presumably meant it was important for logging if something failed, and this version avoids creating the new thread.

Reply:
> After a thread is returned to the ThreadPool, I would assume no one cares about it, right?

If the thread retains the name assigned here after it is returned to the pool it would be pretty confusing. Also, `Thread.Name` is documented to be a write-once property. It should be used only on a newly created thread when the first ones setting it or under an `if (Name == null)` check. The implementation seems to be more permissive, at least on Core but I believe that this line should be removed.

Reply:
I'll remove this, it was kinda cool to see the thread name under the debugger and I didn't see it crash anything but better safe than sorry...